src/python/dxpy/app_builder.py

```diff
@@ -212,58 +161,199 @@ def upload_resources(src_dir, project=None, folder='/', ensure_upload=False):
     if os.path.exists(resources_dir) and len(os.listdir(resources_dir)) > 0:
         target_folder = applet_spec['folder'] if 'folder' in applet_spec else folder
 
-        # Optimistically look for a resource bundle with the same
-        # contents, and reuse it if possible. The resource bundle
-        # carries a property 'resource_bundle_checksum' that indicates
-        # the checksum; the way in which the checksum is computed is
-        # given in the documentation of _directory_checksum.
+        # Copy the code previously in _directory_checksum here and we will create
```

Comment:
Since you removed _directory_checksum, you can remove its mention here. However, please preserve the comment that used to be in that method that specified the way in which the checksum is computed.